### PR TITLE
Handover Artifactory orb source to JFrog

### PR DESCRIPTION
**Pull request by iynere**
- contains all source code from https://github.com/CircleCI-Public/artifactory-orb
- requisite superficial changes have been made (updating links / publishing targets from `circleci/` to `jfrog/`, etc.)

### 1. create an orb called `artifactory` in the `jfrog` orb namespace
- first, create the namespace if it does not yet exist—`circleci namespace create jfrog github jfrog`
- then create the orb:`circleci orb create jfrog/artifactory`, clone the repository, & run the included `publish-alpha.sh` script once locally, to bootstrap the automated testing process
### 2. start building this project on CircleCI
the repo is set up with automated testing / publishing on CircleCI, so it will need to build on CircleCI for that automation to function
### 3. generate & store an SSH key on GitHub & CircleCI
the automated testing flow is dependent on an automated process of pushing a new git tag back to the repository, from within a CircleCI job, which requires generating/storing a public/private keypair with read/write permissions ([see instructions here](https://support.circleci.com/hc/en-us/articles/360021127693-How-to-generate-and-store-read-write-SSH-keys)) & adding the key fingerprint to your `config.yml` (see comments [here](https://github.com/jfrog/artifactory-orb/pull/1/files#r287050983) & [here](https://github.com/jfrog/artifactory-orb/pull/1/files#r287051109))
### 4. recreate demo/testing environment in Artifactory
- automated publishing jobs will not run unless all tests pass. tests require an active jfrog artifactory account.
- you can replicate the setup in our demo account: https://orbdemos.jfrog.io/orbdemos (currently inactive, pinging y'all to reactivate it)
### 5. you will then need to update the related parameter values and environment variables
- parameter values in `config.yml` ([here](https://github.com/jfrog/artifactory-orb/pull/1/files#r287056598), [here](https://github.com/jfrog/artifactory-orb/pull/1/files#r287056911), [here](https://github.com/jfrog/artifactory-orb/pull/1/files#r287057051), [here](https://github.com/jfrog/artifactory-orb/pull/1/files#r287057391), [here](https://github.com/jfrog/artifactory-orb/pull/1/files#r287057557), [here](https://github.com/jfrog/artifactory-orb/pull/1/files#r287057715), [here](https://github.com/jfrog/artifactory-orb/pull/1/files#r287057945), [here](https://github.com/jfrog/artifactory-orb/pull/1/files#r287058063), [here](https://github.com/jfrog/artifactory-orb/pull/1/files#r287058200), [here](https://github.com/jfrog/artifactory-orb/pull/1/files#r287058307), )
- environment variables (`ARTIFACTORY_URL`, `ARTIFACTORY_API_KEY`, `ARTIFACTORY_USER`) stored in the `jfrog/artifactory-orb` project on CircleCI ([instructions](https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-project)) or via a [Context](https://circleci.com/docs/2.0/contexts/) in the `jfrog` organization on CircleCI (in which case, the context will need to be added to any of the CircleCI jobs that require it—[see our example](https://github.com/CircleCI-Public/artifactory-orb/blob/cdfd47da2d7c40519f0f11bac35607173871bfa7/.circleci/config.yml#L87))
### 6. publishing a production version of the orb
once all these steps are complete, a production version of the orb will be published automatically upon a successful (only if all CircleCI jobs pass) commit to master

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+            - orb-tools/lint
+
+      - orb-tools/publish-dev:
+          orb-name: jfrog/artifactory
+          requires:
+            - orb-tools/pack
+
+      - orb-tools/trigger-integration-workflow:
+          name: trigger-integration-dev
+          ssh-fingerprints: # fill in your fingerprint here
```
Comment (iynere) [inline .circleci/config.yml:86]: add fingerprint here

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+          cleanup-tags: true
+          requires:
+            - orb-tools/publish-dev
+          filters:
+            branches:
+              ignore: master
+
+      - orb-tools/trigger-integration-workflow:
+          name: trigger-integration-master
+          ssh-fingerprints: # fill in your fingerprint here
```
Comment (iynere) [inline .circleci/config.yml:96]: add fingerprint here

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+      # test with file spec
+      - artifactory/upload:
+          use-file-spec: true
+          file-spec: spec.json
+          spec-vars: spec-pattern=*.jar;spec-target=orb-testing/all-commands/file-spec/
+
+      - artifactory/build-integration
+
+      - artifactory/docker-login:
+          docker-registry: orbdemos-docker.jfrog.io
```
Comment (iynere) [inline .circleci/config.yml:35]: input your docker registry here

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+      - run:
+          name: build sample image
+          command: |
+            docker build -t \
+              orbdemos-docker.jfrog.io/hello-world-default:1.0-${CIRCLE_BUILD_NUM} \
```
Comment (iynere) [inline .circleci/config.yml:41]: update docker tag here

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+              docker-publish-assets
+
+      - artifactory/docker-publish:
+          docker-tag: orbdemos-docker.jfrog.io/hello-world-default:1.0-${CIRCLE_BUILD_NUM}
```
Comment (iynere) [inline .circleci/config.yml:45]: update docker tag here

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+                  echo "not a jar" > artifact1.jar
+                  echo "not a jar" > artifact2.jar
+          filters: *integration-dev_filters
+
+      - artifactory/docker-publish:
+          name: test-publish-dev
+          docker-registry: orbdemos-docker.jfrog.io
+          repository: docker
+          build-name: hello-alpine
+          docker-tag: orbdemos-docker.jfrog.io/hello-world-default:1.0-${CIRCLE_BUILD_NUM}
```
Comment (iynere) [inline .circleci/config.yml:142]: update docker tag here

File: .circleci/config.yml [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+          pre-steps: [checkout, run: "cp docker-publish-assets/* ."]
+          filters: *integration-dev_filters
+
+      - artifactory/docker-publish:
+          name: test-publish-custom-dev
+          docker-registry: orbdemos-docker.jfrog.io
```
Comment (iynere) [inline .circleci/config.yml:148]: update docker registry here

File: .circleci/config.yml [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+          repository: docker
+          build-name: hello-alpine-custom
+          docker-tag: orbdemos-docker.jfrog.io/hello-alpine-custom:1.0-${CIRCLE_BUILD_NUM}
```
Comment (iynere) [inline .circleci/config.yml:151]: update docker tag here

File: .circleci/config.yml [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+            - run:
+                name: create sample artifact files
+                command: |
+                  echo "not a jar" > artifact1.jar
+                  echo "not a jar" > artifact2.jar
+          filters: *integration-master_filters
+
+      - artifactory/docker-publish:
+          name: test-publish-master
+          docker-registry: orbdemos-docker.jfrog.io
```
Comment (iynere) [inline .circleci/config.yml:190]: update docker registry here

File: .circleci/config.yml [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+          repository: docker
+          build-name: hello-alpine
+          docker-tag: orbdemos-docker.jfrog.io/hello-world-default:1.0-${CIRCLE_BUILD_NUM}
```
Comment (iynere) [inline .circleci/config.yml:193]: update docker tag here

File: .circleci/config.yml [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+          pre-steps: [checkout, run: "cp docker-publish-assets/* ."]
+          filters: *integration-master_filters
+
+      - artifactory/docker-publish:
+          name: test-publish-custom-master
+          docker-registry: orbdemos-docker.jfrog.io
```
Comment (iynere) [inline .circleci/config.yml:199]: update docker registry here

File: .circleci/config.yml [hunk 12]
```diff
@@ -0,0 +1,240 @@
...
+          repository: docker
+          build-name: hello-alpine-custom
+          docker-tag: orbdemos-docker.jfrog.io/hello-alpine-custom:1.0-${CIRCLE_BUILD_NUM}
```
Comment (iynere) [inline .circleci/config.yml:202]: update docker tag here

**Comment by eyalbe4** [comment]
Thank you @iynere!
Can you please join our contributors community by signing [JFrog's CLA](https://secure.echosign.com/public/hostedForm?formid=5IYKLZ2RXB543N) so that we can merge your code?

**Comment by iynere** [comment]
@eyalbe4 this isn't actually my code, i've forked it from https://github.com/CircleCI-Public/artifactory-orb, so i'm not sure if my signing that document is really appropriate or sufficient... let me check in with @ttrahan about this—he or i may follow up over email

**Comment by markgalpin** [comment]
I have taken an action to have Raj follow up through other channels.

**Pull request merged**